pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -36,399 +36,142 @@ If your bot fails to start with a `PrivilegedIntentsRequired` exception, this in
 
-If you don't wish to use the provided `config.yml` above, these are the main sections in `config-default.yml` that need overriding:
+**Note**: Skip this part if you used the automatic configuration.
 
-* `guild.id`
-* `guild.categories`
-* `guild.channels`
-* `guild.roles`
-* `guild.webhooks`
-* `style.emojis`
+##### .env.server
+Reading this means that you're ready for a bit of manual labour.
+If for some reason you've missed the automatic server setup section, you can read about it [here](#automatic-configuration)
+
+To configure the bot manually, you will **only** need to set the values for the channels, roles, categories, etc.
+that are used by the component you are developing.
+
+For example, if we're testing a feature that only needs the `announcements` channel:
+
+`constants.py`
+
+```py
+
+class EnvConfig:
+    # Defines from where & how Pydantic will be looking for env variables
+    ...
+
+class _Channels(EnvConfig):
+
+    EnvConfig.Config.env_prefix = "channels_"
+
+    announcements = 1079790565794779156
+    changelog = 1077877318564991006
+
+# Instantiate the class & load the configuration
+Channels = _Channels()
+```
+
+`.env.server` file:
+
+```text
+# .env.server
+
+channels_announcements=1077875228002234398
+```
+
+When you launch your bot, `pydantic` will load up the server constants from the `.env.server` file if they exist.
+
+Each constants class will define its own prefix, which will make `pydantic` look for variables that will look like `{{env_prefix}}{{attribute_name}}` in the environment files
+
+In our example, this will imply that pydantic will look for both `channels_announcements` and `channels_changelog` in the `.env.server` file.
+
+As you can see here, only `channels_announcements` has been defined in the `.env.server` file since it's the only one needed, which will tell `pydantic`
+to use the value **1077875228002234398** for the `announcements` attribute instead of the default **1079790565794779156**, and use the default value for the `changelog` attribute
+
+```python
+>>> Channels.announcements
+1077875228002234398
+>>> Channels.changelong
+1077877318564991006
+```
+
+See [here](../obtaining-discord-ids) for help with obtaining Discord IDs.
+
+If you wish to set all values in your `env.server` for your testing server, you need to set **all** the ones prefixed with:
+
+* `guild_`
+* `categories_`
+* `channels_`
+* `roles_`
+* `webhooks_`
+* `emojis_`
 
 Additionally:
 
-* At this stage, set `bot.redis.use_fakeredis` to `true`. If you're looking for instructions for working with Redis, see [Working with Redis](#optional-working-with-redis).
-* Set `urls.site_api` to `!JOIN [*DOMAIN, "/api"]`.
-* Set `urls.site_schema` and `urls.site_api_schema` to `"http://"`.
+* At this stage, set `redis_use_fakeredis` to `true`. If you're looking for instructions for working with Redis, see [Working with Redis](#optional-working-with-redis).
+* Set `urls_site_schema` and `urls_site_api_schema` to `"http://"`.
```

Comment:
This part is unrelated to the configuration method, since it's not set automatically by botstrap. We could set it automatically to true if there's no existing config, or move this part to an area common to both methods.

Reply:
The default values of https for urls_site_schema and HTTP for urls_site_api_schema are set properly for dev configurations.

Reply:
Are you sure it doesn't need to be HTTP? it's information from the existing guide (which I didn't edit in the rewrite). If they don't need to be changed, then we can remove mentions of them from the guide altogether, and only move the mention of fakeredis.

Reply:
It only needs to be HTTP if you're using a local site for `urls_site`, which I didn't realize this article was suggesting as well. I think it's more useful for most users to use the prod domain, since the site URL is only used for log viewing and paste. Log viewing is only sent in mod-log, while the pastebin is used in a few locations.

Reply:
This is still relevant. The instruction about fakeredis can be moved under
https://github.com/python-discord/site/blob/4eedb972d4712082b9a2e381ea728a4fd59ed445/pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md?plain=1#L217